mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVaultTest.java

```diff
@@ -50,6 +52,16 @@ public DeletedMessageVault getVault() {
     }
 
 
+    @Disabled("the BlobStoreDeletedMessageVaultTest.append(user, deletedMessage, inputStream)" +
+        "doesn't use user argument, it use the owner retrieved from deletedMessage for storing messages" +
+        "this test passes not the same user and owner in DeletedMessage to append()" +
+        "as the consequence, the filter cannot rely on user argument" +
+        "It raised a concern in append() where we pass user information in two different arguments")
+    @Test
+    @Override
+    public void searchForAnUserShouldNotReturnMessagesFromAnotherUser() {
```

Comment:
A quick debug shows that

```
user: User{localPart=dimitri, domainPart=Optional[Domain : apache.org]}

 AND 

deletedMessageUser User{localPart=bob, domainPart=Optional[Domain : apache.org]}
```

Well that is a concern... This should be forbidden to store Dimitri message in bob vault, isn't it?

Shouldn't we rather reject that operation? Or better: modify the DeletedMessageVault API to remove that duplication and make the API less dangerous to play with:

```
public interface DeletedMessageVault {
    Publisher<Void> append(DeletedMessage deletedMessage, InputStream mimeMessage);
}
```

This clearly seems like the way to go...